vmdb/app/models/host_vmware_esx.rb

```diff
@@ -76,13 +75,24 @@ def vim_enable_vmotion(device = nil)
 
   def vim_disable_vmotion(device = nil)
     with_provider_object do |vim_host|
-      vnm      = vim_host.hostVirtualNicManager
-      device ||= vnm.candidateVnicsByType("vmotion").first.device rescue nil
+      vnm, device = get_host_virtual_nic_manager_with_vmotion_device(vim_host, device)
       $log.info "MIQ(Host.vim_disable_vmotion) Invoking for device=<#{device}>"
       vnm.deselectVnicForNicType("vmotion", device) unless device.nil?
     end
   end
 
+  def get_host_virtual_nic_manager_with_vmotion_device(vim_host, device = nil)
+    vnm = vim_host.hostVirtualNicManager
+    begin
+      device ||= vnm.candidateVnicsByType("vmotion").first.device
+    rescue
+      # TODO: specify what to rescue
+      # TODO: log it
+      nil
+    end
```

Comment:
Again, I'd prefer the `rescue nil`, though I'm curious about why it was there in the first place.  @gmcculloug Do you know?